src/validator/MatchValidator.cpp

```diff
@@ -694,44 +695,68 @@ Status MatchValidator::validateGroup(YieldClauseContext &yieldCtx) const {
     for (auto *col : cols) {
         auto *colExpr = col->expr();
         auto colOldName = col->name();
-        if (colExpr->kind() != Expression::Kind::kAggregate) {
-            auto collectAggCol = colExpr->clone();
-            auto aggs =
-                ExpressionUtils::collectAll(collectAggCol, {Expression::Kind::kAggregate});
-            for (auto *agg : aggs) {
-                DCHECK_EQ(agg->kind(), Expression::Kind::kAggregate);
-                if (!ExpressionUtils::checkAggExpr(static_cast<const AggregateExpression *>(agg))
-                         .ok()) {
-                    return Status::SemanticError("Aggregate function nesting is not allowed: `%s'",
-                                                 colExpr->toString().c_str());
+        auto aggs = ExpressionUtils::collectAll(colExpr, {Expression::Kind::kAggregate});
+        for (auto *agg : aggs) {
+            DCHECK_EQ(agg->kind(), Expression::Kind::kAggregate);
+            if (!ExpressionUtils::checkAggExpr(static_cast<const AggregateExpression *>(agg))
+                     .ok()) {
+                return Status::SemanticError("Aggregate function nesting is not allowed: `%s'",
+                                             colExpr->toString().c_str());
+            }
+        }
+    }
+
+    yieldCtx.needGenProject_ = true;
+    for (auto *col : cols) {
+        auto *colExpr = col->expr();
+        auto colOldName = col->toString();
+        auto foldRes = ExpressionUtils::foldConstantExpr(colExpr);
+        NG_RETURN_IF_ERROR(foldRes);
+        auto *newExpr = foldRes.value();
```

Comment:
You only need to do some check here(maybe use `ExpressionUtils::isEvaluableExpr`), it is better for the optimizer to do constant folding.